libkpod/image/copy.go

```diff
@@ -100,8 +104,115 @@ func PushImage(srcName, destName string, options CopyOptions) error {
 	return nil
 }
 
+type imagePullStruct struct {
+	imageName   string
+	tag         string
+	registry    string
+	hasRegistry bool
+	transport   string
+}
+
+func (ips imagePullStruct) returnFQName() string {
+	fqname := fmt.Sprintf("%s%s/%s:%s", ips.transport, ips.registry, ips.imageName, ips.tag)
+	return fqname
+}
+
+func imageBeginsWithTransport(image string) bool {
+	parts := strings.SplitN(image, ":", 2)
+	if len(parts) == 1 {
+		return false
+	}
+	for _, transport := range transports.ListNames() {
+		if strings.HasPrefix(parts[0], transport) {
+			return true
+		}
+	}
+	return false
+}
+
+func getRegistriesToTry(image string) ([]types.ImageReference, error) {
+	var registries []types.ImageReference
+	var imageError = fmt.Sprintf("Unable to parse '%s'\n", image)
+	imgRef, err := reference.Parse(image)
+	if err != nil {
+		return registries, errors.Errorf(imageError)
+
+	}
+	tagged, istagged := imgRef.(reference.NamedTagged)
+	tag := "latest"
+	if istagged {
+		tag = tagged.Tag()
+	}
+	hasDomain := true
+	registry := reference.Domain(imgRef.(reference.Named))
+	if registry == "" {
+		hasDomain = false
+	}
+	imageName := reference.Path(imgRef.(reference.Named))
+	pImage := imagePullStruct{
+		imageName,
+		tag,
+		registry,
+		hasDomain,
+		DefaultRegistry,
+	}
+
+	if pImage.hasRegistry {
+		// If input has a registry, we have to assume they included an image
+		// name but maybe not a tag
+		pullRef, err := alltransports.ParseImageName(pImage.returnFQName())
+		if err != nil {
+			return registries, errors.Errorf(imageError)
+		}
+		registries = append(registries, pullRef)
+	} else {
+		// No registry means we check the globals registries configuration file
+		// and assemble a list of candidate sources to try
+
+		// Should I be catching the case here where there are no searchable registries
```

Comment:
Please remove this line once the question is answered.   FWIW, I think you should be bailing here as you have it below, but will defer to opinions from other folk.

Reply:
Agree with bailing if we can't find a usable registry. Seems like the thing users would expect to happen.

Reply:
comment leftover, gone